python/cudf/cudf/core/buffer.py

```diff
@@ -72,23 +71,48 @@ def __init__(
                 raise TypeError("data must be Buffer, array-like or integer")
             self._init_from_array_like(np.asarray(data), owner)
 
+    @classmethod
+    def from_buffer(cls, buffer: Buffer, size: int = None, offset: int = 0):
```

Comment:
We could consider a slice API instead of a classmethod factory here. It might be more Pythonic?

Before:
```python
buf = Buffer.from_buffer(
    buffer=codes_column.base_data,
    size=self.size * codes_column.dtype.itemsize,
    offset=self.offset * codes_column.dtype.itemsize,
)
```
After:
```python
start_bytes = self.offset * codes_column.dtype.itemsize
end_bytes = start_bytes + self.size * codes_column.dtype.itemsize
buf = codes_column.base_data[start_bytes:end_bytes]
```

Suggested implementation:
```python
def __getitem__(self, key):
    if not isinstance(key, slice):
        raise ValueError("Buffers can be sliced but individual items cannot be accessed.")
    start, stop, step = key.start, key.stop, key.step
    if step != 1:
        raise ValueError("A Buffer can only be sliced with step = 1.")
    if start < 0:
        start += len(self)
    if stop < 0:
        stop += len(self)
    if stop > len(self) or start < 0:
        raise ValueError("The slice start and stop must be in bounds.")
    if start > stop:
        raise ValueError("The slice start cannot be greater than the slice stop.")
    ret = cls()
    ret._ptr = self._ptr + start
    ret.size = stop - start
    ret._owner = self
    return ret
```